src/Azure.DataApiBuilder.Mcp/BuiltInTools/DescribeEntitiesTool.cs

```diff
@@ -48,107 +69,319 @@ public Task<CallToolResult> ExecuteAsync(
+                        try
                         {
-                            List<object> fields = new();
-
-                            // Try to get stored procedure parameters if available
-                            if (entity.Source.Parameters != null)
-                            {
-                                foreach (KeyValuePair<string, object> parameter in entity.Source.Parameters)
-                                {
-                                    fields.Add(new
-                                    {
-                                        name = parameter.Key,
-                                        // TODO: "type" and "description"
-                                    });
-                                }
-                            }
+                            Dictionary<string, object?> entityInfo = nameOnly
+                                ? BuildBasicEntityInfo(entityName, entity)
+                                : BuildFullEntityInfo(entityName, entity);
 
-                            entityInfo["fields"] = fields;
+                            entityList.Add(entityInfo);
+                        }
+                        catch (Exception ex)
+                        {
+                            logger?.LogWarning(ex, "Failed to build info for entity {EntityName}", entityName);
                         }
+                    }
+                }
 
-                        // Extract permissions from entity configuration
-                        List<string> permissions = new();
+                if (entityList.Count == 0)
+                {
+                    if (entityFilter != null && entityFilter.Count > 0)
+                    {
+                        return Task.FromResult(McpResponseBuilder.BuildErrorResult(
+                            "EntitiesNotFound",
+                            $"No entities found matching the filter: {string.Join(", ", entityFilter)}",
+                            logger));
+                    }
+                    else
+                    {
+                        return Task.FromResult(McpResponseBuilder.BuildErrorResult(
+                            "NoEntitiesConfigured",
+                            "No entities are configured in the runtime configuration.",
+                            logger));
+                    }
+                }
+
+                cancellationToken.ThrowIfCancellationRequested();
+
+                entityList = entityList.OrderBy(e => e["name"]?.ToString() ?? string.Empty).ToList();
 
-                        if (entity.Permissions != null)
+                List<object> finalEntityList = entityList.Cast<object>().ToList();
+
+                Dictionary<string, object?> responseData = new()
+                {
+                    ["entities"] = finalEntityList,
+                    ["count"] = finalEntityList.Count,
+                    ["mode"] = nameOnly ? "basic" : "full"
```

Comment:
The mode in the response was not expected as per the spec. Why do we need to add it? Without looking at code, I didnt realize basic meant nameOnly. We should remove this from the response unless we confirm value from adding this.